File: models/pool.py
from dataclasses import dataclass
from datetime import datetime
from typing import Optional

@dataclass
class Pool:
    pool_code: str
    league_code: str
    season: str
    division_id: str
    league_name: str
    raw_name: str
    name: str
    short_name: str
    format: str
    gender: str
    followers_count: Optional[int] = 0
    active: bool = True
    id: Optional[int] = None
    created_at: Optional[datetime] = None
    last_update: Optional[datetime] = None

File: models/Pool.ts
export enum PoolDivisionCode {
    REG = "REG",
    NAT = "NAT",
    PRO = "PRO",
}

export enum PoolGender {
    M = "M",
    F = "F",
}

export interface Pool {
    id: number;
    pool_code: string;
    league_code: string;
    season: number;
    division_code: PoolDivisionCode;
    gender?: PoolGender;
    league_name?: string;
    pool_name?: string;
    division_name?: string;
    raw_division_name?: string;
    last_update?: Date;
    active: boolean;
}